common/element_handle.go

```diff
@@ -64,7 +64,7 @@ type (
 )
 
 func getElementHandleActionFn(
-	h *ElementHandle, states []string, fn elementHandleActionFn, force, noWaitAfter bool, timeout time.Duration,
+	elh *ElementHandle, states []string, ehaFn elementHandleActionFn, force, noWaitAfter bool, timeout time.Duration,
```

Comment:
I'm against varnamelen linter and on the side of reverting these changes. There is nothing wrong accepting `h` or `fn`. It is idiomatic to have these as short parameters. Their types make it clear what's going on.

Reply:
I'm not sure... I see the benefit of `varnamelen`, even in this case. The scope of this function is too large to use vars like `h` and `fn`. This is fine for ~5 LOC functions or `for` loops, but not here.

We definitely abuse this often, as you can see from the 339 errors (not sure how you got 1k?). But I think most of these can be quickly changed using `sed` or [`rf`](https://pkg.go.dev/rsc.io/rf). And we can always [tweak the configuration](https://github.com/blizzy78/varnamelen#golangci-lint-integration) to e.g. relax the scope requirements.

So I'm leaning towards keeping it, as it's valuable with something we're likely to not point out in PRs.

Reply:
Sorry, I see no benefit here, and I'm firmly against it. This is not something we want to enforce on senior devs. Better use linters for practical purposes instead of forcing non-idiomatic style changes.

Reply:
I just ran this useless linter on Go stdlib and it found 14K warnings. Come on.

Reply:
I just ran the linter on Go stdlib and it found 14K warnings. Come on.

Reply:
It's not an issue of seniority :smile:, but properly naming vars according to the scope they're used in. It's about legibility, avoiding shadowing, `grep`ability, etc.

I disagree, but if you feel strongly about it, sure, I'll remove it.

Reply:
Please remove it 🙏 

Btw, Rob won't agree. I ran the linter on his ivy project: 300 warnings. It seems he can't write idiomatic code according to this linter...

Reply:
Ah, I certainly wouldn't want to be maintaining his ivy project then :stuck_out_tongue:

FWIW, I prefer not blindly sticking to whatever is "idiomatic", but to whatever improves the DX while working on the code. To me it's a no brainer that longer var names _for larger scopes_ are the better approach.

Reply:
> longer var names for larger scopes are the better approach

I agree and but it depends on a blend of a dev's experience + code at hand. An automatic tool cannot decide that. When I recommend something, it's because I know it from my experience, not because I follow people blindly.